VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/Utilities/ExtensionsTests.cs

```diff
@@ -99,5 +99,148 @@ void Method(string input)
+            }
+    }";
+
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+            var compilation = CSharpCompilation.Create("MyCompilation", new[] { tree }, new[] { mscorlib });
+            var semanticModel = compilation.GetSemanticModel(tree);
+
+            var root = tree.GetRoot();
+            var method = root.DescendantNodes().OfType<MethodDeclarationSyntax>().First();
+            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
+            Assert.IsTrue(methodSymbol.IsAsync());
+        }
+
+        [TestMethod]
+        public void IsAsync_WithGenericTaskReturnType()
+        {
+            var source = @"
+    using System;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            Task<int> Method()
+            {
+                return Task.FromResult(0);
+            }
+    }";
+
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+            var compilation = CSharpCompilation.Create("MyCompilation", new[] { tree }, new[] { mscorlib });
+            var semanticModel = compilation.GetSemanticModel(tree);
+
+            var root = tree.GetRoot();
+            var method = root.DescendantNodes().OfType<MethodDeclarationSyntax>().First();
+            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
+            Assert.IsTrue(methodSymbol.IsAsync());
+        }
+
+        [TestMethod]
+        public void IsAsync_WithAsyncKeywordAndVoidReturnType()
+        {
+            var source = @"
+    using System;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            async void Method()
+            {
+            }
+    }";
+
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+            var compilation = CSharpCompilation.Create("MyCompilation", new[] { tree }, new[] { mscorlib });
+            var semanticModel = compilation.GetSemanticModel(tree);
+
+            var root = tree.GetRoot();
+            var method = root.DescendantNodes().OfType<MethodDeclarationSyntax>().First();
+            var methodSymbol = semanticModel.GetDeclaredSymbol(method);
+            Assert.IsTrue(methodSymbol.IsAsync());
+        }
+
+        [TestMethod]
+        public void IsAsync_WithVoidReturnType()
+        {
+            var source = @"
+    using System;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    namespace ConsoleApplication1
+    {
+        class MyClass
+        {
+            void Method()
+            {
+            }
+    }";
+
+            var tree = CSharpSyntaxTree.ParseText(source);
```

Comment:
This piece of code to create a compilation and check the method is often repeated. Can you factor this out to a separate method `CheckAsync(boolean)` or something like that?